lib/foreman_puppet/register.rb

```diff
@@ -59,89 +59,89 @@
-                                              lookup_values: [:create],
-                                              'foreman_puppet/api/v2/smart_class_parameters': [:create],
-                                              'foreman_puppet/api/v2/override_values': [:create] },
+    permission :puppet_create_external_parameters, { 'foreman_puppet/puppetclass_lookup_keys': %i[new create],
+                                                     lookup_values: [:create],
+                                                     'foreman_puppet/api/v2/smart_class_parameters': [:create],
+                                                     'foreman_puppet/api/v2/override_values': [:create] },
       resource_type: 'ForemanPuppet::PuppetclassLookupKey'
-    permission :edit_external_parameters, { 'foreman_puppet/puppetclass_lookup_keys': %i[edit update],
-                                            lookup_values: %i[create update destroy],
-                                            'foreman_puppet/api/v2/smart_class_parameters': [:update],
-                                            'foreman_puppet/api/v2/override_values': %i[create update destroy] },
+    permission :puppet_edit_external_parameters, { 'foreman_puppet/puppetclass_lookup_keys': %i[edit update],
+                                                   lookup_values: %i[create update destroy],
+                                                   'foreman_puppet/api/v2/smart_class_parameters': [:update],
+                                                   'foreman_puppet/api/v2/override_values': %i[create update destroy] },
       resource_type: 'ForemanPuppet::PuppetclassLookupKey'
-    permission :destroy_external_parameters, { 'foreman_puppet/puppetclass_lookup_keys': [:destroy],
-                                               lookup_values: [:destroy],
-                                               'foreman_puppet/api/v2/smart_class_parameters': [:destroy],
-                                               'foreman_puppet/api/v2/override_values': %i[create update destroy] },
+    permission :puppet_destroy_external_parameters, { 'foreman_puppet/puppetclass_lookup_keys': [:destroy],
+                                                      lookup_values: [:destroy],
+                                                      'foreman_puppet/api/v2/smart_class_parameters': [:destroy],
+                                                      'foreman_puppet/api/v2/override_values': %i[create update destroy] },
       resource_type: 'ForemanPuppet::PuppetclassLookupKey'
   end
 
   security_block :puppet_environments do
-    permission :view_environments, { 'foreman_puppet/environments': %i[index show auto_complete_search welcome],
-                                     'foreman_puppet/api/v2/environments': %i[index show] },
+    permission :puppet_view_environments, { 'foreman_puppet/environments': %i[index show auto_complete_search welcome],
```

Comment:
Do we need db migrations to change the permission names?

Reply:
On second thought, do we want to change the name here? I'm wondering if the `puppet` prefix breaks things or not.

Reply:
Permissions test for master:
https://github.com/ATIX-AG/foreman_puppet/runs/2882491673?check_suite_focus=true

To me it looks like there is a name clash between original permissions and the plugin and the original permissions are overwritten: https://github.com/theforeman/foreman/blob/3f7804fd96d380010b340f2a954f9cb42a60c551/app/registries/foreman/plugin.rb#L330 
That is why I tried renaming. Correct me if I am wrong.

Reply:
But maybe overwriting the existing permission is exactly what we want and we should let the tests fail for Foreman < 3.0 ?

Reply:
Yep, sounds good. With the plugin we do want to override core functionality, so I'd be ok with that.

Reply:
I have changed the PR to contain the permissions check starting from a 3.X stable branch.